clustered/server/src/main/java/org/ehcache/clustered/server/internal/messages/EhcacheMessageTrackerMessage.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.server.internal.messages;
+
+import org.ehcache.clustered.common.internal.messages.EhcacheEntityMessage;
+import org.ehcache.clustered.common.internal.messages.EhcacheEntityResponse;
+import org.terracotta.client.message.tracker.OOOMessageHandler;
+import org.terracotta.entity.ClientSourceId;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Message sending messages that are tracked for duplication. If a passive becoming active receives
+ * a duplicate, it needs to discard it.
+ */
+public class EhcacheMessageTrackerMessage extends EhcacheSyncMessage {
+
+  private final Map<ClientSourceId, Map<Long, EhcacheEntityResponse>> trackedMessages;
```

Comment:
If we keep the `ClientSourceId`s as longs inside the message and delay conversion to ClientSourceId until inside the invoke then we can use the context to reconstruct the `ClientSourceId` instances and avoid creating our own implementation.